File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int zeros=0,l=0,r=0,m=0;
        while(r<nums.length){
            if(nums[r]==0){
                zeros++;
            }
            while(zeros>k){
                if(nums[l]==0){
                    zeros--;
                }
                l++;
            }
            m=Math.max(m,r-l+1);
            r++;
        }
        return m;
    }
}

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        l,r,maxLen,zeros=0,0,0,0
        while(r<len(nums)):
            if nums[r]==0:
                zeros+=1
            if zeros<=k:
                maxLen=max(maxLen,r-l+1)
            if zeros>k:
                if nums[l]==0:
                    zeros-=1
                l+=1
            r+=1
        return maxLen
        